docs/project/list-of-diagnostics.md

```diff
@@ -108,3 +108,14 @@ The diagnostic id values reserved for .NET Libraries analyzer warnings are `SYSL
 |  __`SYSLIB1027`__ | *_`SYSLIB1024`-`SYSLIB1029` reserved for logging._* |
 |  __`SYSLIB1028`__ | *_`SYSLIB1024`-`SYSLIB1029` reserved for logging._* |
 |  __`SYSLIB1029`__ | *_`SYSLIB1024`-`SYSLIB1029` reserved for logging._* |
+
+## Package Support Errors
+
+The id values reserved for .NET Library packages which error on unsupported platforms are `SYSLIB9000` through `SYSLIB9999`.
+
+### Package Error codes (`SYSLIB9000` - `SYSLIB9999`)
```

Comment:
I don't think I expect to have as much id values for this, but this is just following what we've done with other ranges. 

cc: @GrabYourPitchforks

Reply:
I would recommend _not_ using a diagnostic code for this. The purpose of the diagnostic codes is to accompany the [better obsoletions](https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/better-obsoletion.md) design, which allows developers to suppress obsoletion warnings specific to method families without suppressing all obsoletion warnings application-wide.

In this case, we're producing hard errors, not warnings. There's no ability for the developer to suppress anything. I would instead use a standard `<Error Text="..." />` task, include an fwlink as part of the error message (_"... doesn't support \<framework\>. Consider updating .... See \<fwlink\> for more information."_), and not use a custom error code.

Reply:
System.Speech already shipped with SYSLIB9000 as the error code from the release/5.0 branch: https://github.com/dotnet/runtime/blob/25d20e91c44ddc0b22e1c306d354057c2282b4f1/src/libraries/System.Speech/pkg/build/System.Speech.targets#L5.

I was mostly trying to be consistent with System.Speech. from @danmoseley in his PR:

> I included an error code as it appears in the Error List and it's important to have something to search for. I picked SYSLIB9000 as we've used SYSLIB0001-SYSLIB0012 for the compatibility analyzer so it seems a reasonable block to claim for this kinds of error.

Reply:
IMO _System.Speech_ using its own error code is a misdesign which should be reverted in the main branch. We should not propagate this misdesign to new APIs.

Reply:
I figured that error codes are nice for web searches. No reason it has to be SYSLIB* though, I just made that up. I don't feel strongly: I think it's fine to have no error code and as the string is fixed, it's easy to search for.

It's potentially useful we could have an aka.ms link in the text, but again I don't think it's particularly necessary.

Reply:
I removed the error code and rephrased the error message to recommend to upgrade to netcoreapp3.1 or higher.